core/config/project_settings.cpp

```diff
@@ -799,7 +795,6 @@ Error ProjectSettings::_save_settings_binary(const String &p_file, const Map<Str
 		int len;
 		err = encode_variant(p_custom_features, nullptr, len, false);
 		if (err != OK) {
-			memdelete(file);
```

Comment:
Now that we don't have to `memdelete`, these patterns could be changed to use `ERR_FAIL_COND_V(err != OK, err);` so we also benefit from the `unlikely()` use in the macro.

But might also be scope creep for this PR, I could do this in a follow-up while re-reviewing the merged diff.